File: api/brapi/searchSymbol.js
const axios = require("axios");

const BASE_URL = "https://brapi.dev/api";

const makeRequest = async (endpoint, params) => {
  const apiKey = process.env.BRAPI_API_KEY; // Use process.env for Vercel
  if (apiKey) {
    params["token"] = apiKey;
  }

  const url = `${BASE_URL}/${endpoint}`;
  console.log(`Making request to Brapi API: ${url}`, {params}); // Replace logger.info
  try {
    const response = await axios.get(url, {params});
    return response.data;
  } catch (error) {
    console.error("Error making request to Brapi API:", error); // Replace logger.error
    throw new Error("Error making request to Brapi API", { cause: error });
  }
};

module.exports = async (req, res) => {
  // Set CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*'); // Allow all origins for development
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS'); // Allow POST and OPTIONS methods
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type'); // Allow Content-Type header

  // Handle preflight OPTIONS request
  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method Not Allowed' });
  }

  const symbol = req.body.symbol;

  console.log("searchSymbol function called with data:", req.body);
  if (!process.env.BRAPI_API_KEY) { // Use process.env for Vercel
    console.error("BRAPI_API_KEY is not set.");
    return res.status(500).json({ error: "BRAPI_API_KEY is not configured." });
  }

  if (!symbol) {
    return res.status(400).json({
      error: "The function must be called with one argument 'symbol'.",
    });
  }

  const endpoint = "quote/list";
  const params = {
    search: symbol,
    limit: 10,
  };

  try {
    const results = await makeRequest(endpoint, params);
    console.log("searchSymbol function results:", results);
    res.status(200).json({ data: results.stocks || [] });
  } catch (error) {
    console.error("Error in searchSymbol Vercel function:", error);
    res.status(500).json({ error: error.message, details: error.cause?.message });
  }
};

File: api/brapi/searchSymbol.py
import requests
import os
from http.server import BaseHTTPRequestHandler
from api._utils import send_cors_preflight, send_json_response, send_error_response, get_request_body

BASE_URL = "https://brapi.dev/api"

def make_request(endpoint, params):
    """Make request to Brapi API"""
    api_key = os.environ.get('BRAPI_API_KEY')
    if api_key:
        params['token'] = api_key
    
    url = f"{BASE_URL}/{endpoint}"
    print(f"Making request to Brapi API: {url}", {"params": params})
    
    try:
        response = requests.get(url, params=params)
        response.raise_for_status()
        return response.json()
    except Exception as error:
        print(f"Error making request to Brapi API: {error}")
        raise Exception("Error making request to Brapi API") from error

class handler(BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        """Handle CORS preflight request"""
        send_cors_preflight(self, 'POST, OPTIONS')
    
    def do_POST(self):
        """Handle POST request"""
        try:
            # Check for API key
            if not os.environ.get('BRAPI_API_KEY'):
                print("BRAPI_API_KEY is not set.")
                send_json_response(
                    self,
                    {'error': 'BRAPI_API_KEY is not configured.'},
                    status_code=500,
                    methods='POST, OPTIONS'
                )
                return
            
            # Read request body
            data = get_request_body(self)
            
            symbol = data.get('symbol')
            
            print("searchSymbol function called with data:", data)
            
            if not symbol:
                send_json_response(
                    self,
                    {'error': "The function must be called with one argument 'symbol'."},
                    status_code=400,
                    methods='POST, OPTIONS'
                )
                return
            
            endpoint = "quote/list"
            params = {
                'search': symbol,
                'limit': 10,
            }
            
            # Make the request
            results = make_request(endpoint, params)
            print("searchSymbol function results:", results)
            
            # Send success response
            stocks = results.get('stocks', [])
            send_json_response(self, {'data': stocks}, methods='POST, OPTIONS')
            
        except Exception as error:
            print(f"Error in searchSymbol Vercel function: {error}")
            send_error_response(self, error, methods='POST, OPTIONS')
